network-config-analyzer/CalicoPolicyYamlParser.py

```diff
@@ -623,8 +624,11 @@ def parse_policy(self):
         res_policy.policy_kind = NetworkPolicy.get_policy_type_from_dict(self.policy)
 
         policy_spec = self.policy['spec']
-        self._set_affects_ingress_egress(policy_spec, is_profile, res_policy)
-        self._set_selected_peers(policy_spec, is_profile, res_policy)
+        self._set_affects_ingress_egress(policy_spec, is_profile, is_global_np, res_policy)
+        res_policy.selected_peers = self._get_selected_peers(policy_spec, is_profile, res_policy.name)
+        ns_selector = policy_spec.get('namespaceSelector')
+        if ns_selector:
+            res_policy.selected_peers &= self._parse_label_selector(ns_selector, policy_spec, namespace_selector=True)
```

Comment:
Maybe this should be within `_get_selected_peers` : this method is supposed to return the peers selected by the policy, which should also take into account the `namespaceSelector`.